xml/issue3077.xml

```diff
@@ -0,0 +1,62 @@
+<?xml version='1.0' encoding='utf-8' standalone='no'?>
+<!DOCTYPE issue SYSTEM "lwg-issue.dtd">
+
+<issue num="3077" status="New">
+<title><tt>(push|emplace)_back</tt> should invalidate the <tt>end</tt> iterator</title>
+<section><sref ref="[vector.modifiers]"/></section>
+<submitter>Casey Carter</submitter>
+<date>10 Mar 2018</date>
+<priority>99</priority>
+
+<discussion>
+<p>
+<sref ref="[vector.modifiers]"/> paragraph 1 specifies that <tt>emplace_back</tt>
+and <tt>push_back</tt> do not invalidate iterators before the insertion point when
+reallocation is unnecessary:
+<blockquote>
+Remarks: Causes reallocation if the new size is greater than the old capacity.
+Reallocation invalidates all the references, pointers, and iterators referring to
+the elements in the sequence. If no reallocation happens, all the iterators and
+references before the insertion point remain valid. [&hellip;]
+</blockquote>
+This statement is redundant, given the blanket wording in
+<sref ref="[container.requirements.general]"/> paragraph 12:
+<blockquote>
+Unless otherwise specified (either explicitly or by defining a function in terms
+of other functions), invoking a container member function or passing a container
+as an argument to a library function shall not invalidate iterators to, or change
+the values of, objects within that container.
+</blockquote>
+It seems that this second sentence (1) should be a note that reminds us that the
+blanket wording applies here when no reallocation occurs, and/or (2) actually
+intends to specify that iterators at and after the insertion point are invalidated.

```

Comment:
I think it's meant to do this and we should rephrase it accordingly rather than making it a note.

Reply:
The PR turns this sentence into a note to accomplish purpose two, and introduces new wording for purpose one. (I see no need to invalidate iterators *after* the end, which I believe cannot be valid on any case.)

Reply:
This sentence is in a paragraph that applies to all insertion operations, including `insert` and `emplace`, not just `push/emplace_back`. The first two can insert in the middle, in which case we need to invalidate everything at or after the insertion point.

Reply:
Yes, thanks - I'll fix this.